solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -1484,6 +1492,60 @@ parameter_types! {
     pub const MaxLengthParaIds: u32 = 100u32;
     pub const MaxEncodedGenesisDataSize: u32 = 5_000_000u32; // 5MB
 }
+
+pub struct InnerStarlightRegistrar<AccountId, RegistrarManager>(
+    PhantomData<(AccountId, RegistrarManager)>,
+);
+impl<AccountId, RegistrarManager> RegistrarHandler<AccountId>
+    for InnerStarlightRegistrar<AccountId, RegistrarManager>
+where
+    RegistrarManager: RegistrarInterface<AccountId = AccountId>,
+{
+    fn register(
+        who: AccountId,
+        id: ParaId,
+        genesis_storage: Vec<ContainerChainGenesisDataItem>,
+    ) -> DispatchResult {
+        // Build HeadData
+        let key_values: Vec<(Vec<u8>, Vec<u8>)> =
+            genesis_storage.into_iter().map(|x| x.into()).collect();
+        let genesis_head = HeadData(key_values.clone().encode());
```

Comment:
This HeadData looks wrong, it is supposed to be a block header, right? Or does the genesis head data include all the genesis storage on chain? `max_head_data_size` is configurable but the max value is 1MB, which is smaller than our limit of 5MB in `MaxGenesisDataSize`, so if genesis_head is the genesis state and not a hash, this can cause problems.

Also careful when cloning the genesis storage as it can be huge.

Reply:
The genesis head is supposed to include all the genesis storage on chain yes, so if the limits are different, we may encounter problems there. I'll be thinking of some solutions.

Reply:
About the clone, at the end it's not necessary here. By removing it everything compiles well.

Reply:
_fixed_

Reply:
At the end the `HeadData` is supposed to be only the genesis header, not the whole state. Thus, we decided to introduce a new param `head_data` to associated `register` functions of `ContainerRegistrar` pallet. This is because otherwise, without this param it would be really difficult to calculate the needed `HeadData` object (and do it properly).